src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs

```diff
@@ -8507,6 +8507,308 @@ public static ulong Splice(ulong[] first, ulong[] second, ulong[] maskArray, int
             return result;
         }
 
+        private static T ConditionalSelectResult<T>(T maskResult, T result, T falseResult) where T : INumberBase<T>
+        {
+            return (maskResult != T.Zero) ? result : falseResult;
+        }
+
+        private static T ConditionalSelectTrueResult<T>(T maskResult, T result, T trueResult) where T : INumberBase<T>
```

Comment:
>ConditionalSelectTrueResult [](http://example.com/codeflow?start=25&length=27)

I don't see place where we are validating this requirement:

>  Inactive elements will not cause a read from Device memory or signal a fault, and are set to zero in the destination vector.

"and are set to zero in the destination vector.".

Reply:
its ok to add this in a follow-up PR.

Reply:
I think a follow-up PR would be fine. We should do this for all test cases that use ConditionalSelectTrueResult.